server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java

```diff
@@ -147,6 +155,12 @@ public void onNext(DiscoveryRequest request) {
             nonce,
             request.getVersionInfo());
 
+        // If the onStreamRequest method returns false then we should ignore this request.
```

Comment:
curious what the intended use case for this is? how will envoy behave if we ignore it's request?

Reply:
That was part of a half-baked plan to give consumers the ability to define custom logic delaying responses while waiting for data to exist, or custom handling of errors to avoid situations where envoy ends up in the infinite disco request loop. I just pushed a new commit that backs that part out.